### PR TITLE
Adding section and a book for Blazor Framework

### PR DESCRIPTION
**Pull request by madhavsarpal**
## Hacktoberfest notes:

- due to volume of submissions, we may not be able to review PRs that do not pass tests and do not have informative titles.
- please read our [contributing guidelines](/CONTRIBUTING.md)
- be sure to check the output of Travis-CI for linter errors
- if this is your first open source contribution, make sure it's not your last!

## What does this PR do?
Add Resource

### Description 
Adding a section and a book for Blazor Framework

### Why is this valuable (or not)?
Blazor is a new client side framework by Microsoft meant to replace older technologies like ASP.NET web forms. This book provides an introduction to Blazor Framework

### How do we know it's really free?
It is available for free download on Microsoft's website and can be downloaded without signing-up for anything

### For book lists, is it a book? For course lists, is it a course? etc.
It is a Book

### Checklist:
- [x] Not a duplicate
- [x] Included author(s) if appropriate
- [x] Lists are in alphabetical order
- [x] Needed indications added (PDF, access notes, under construction)

### CODE REVIEW

File: free-programming-books.md [hunk 1]
```diff
@@ -1025,6 +1026,11 @@ Kerridge (PDF) (email address *requested*, not required)
 * [Object-Oriented Programming in the BETA Programming Language](https://beta.cs.au.dk/Books/) - Ole Lehrmann Madsen, Birger Møller-Pedersen, Kristen Nygaard
 
 
+### Blazor
+
+* [Blazor for ASP.NET Web Forms Developers](https://dotnet.microsoft.com/download/e-book/blazor-for-web-forms-devs/pdf) - Daniel Roth, Jeff Fritz, Taylor Southwick
```
Comment (SethiPandi) [inline free-programming-books.md:1031]: As this leads to a PDF download, could you please append (PDF) to the end of the line?

```suggestion
* [Blazor for ASP.NET Web Forms Developers](https://dotnet.microsoft.com/download/e-book/blazor-for-web-forms-devs/pdf) - Daniel Roth, Jeff Fritz, Taylor Southwick (PDF)
```

**Review by SethiPandi — CHANGES_REQUESTED** [review]
Just a small comment!

You can get more guidance on the formatting and conventions of this repository by checking out the [contribution guide](https://www.github.com/EbookFoundation/free-programming-books/tree/master/CONTRIBUTING.md).

**Comment by eshellman** [comment]
would more people find this if it was in the asp.net section?

**Comment by madhavsarpal** [comment]
> would more people find this if it was in the asp.net section?

But Blazor is a separate technology of it own, and although the book draws paralellism between ASP.NET web forms and Blazor, it talks about Blazor separately too and the focus is on Blazor and not ASP.NET. Infact, we can use Blazor without ASP.NET with any other backend technology so I though it would be more appropriate to have a new section for that.

**Pull request merged**